commit 89360b87c3efa5d265a3b43dd8fc245c8f4356e3
Author: Vishal Verma <vishal.l.verma@intel.com>
Date:   Mon Oct 30 13:22:21 2017 -0700

    libnvdimm, badrange: remove a WARN for list_empty
    
    Now that we're reusing the badrange functions for nfit_test, and that
    exposes badrange injection/clearing to userspace via the DSM paths, it
    is plausible that a user may call the clear DSM multiple times. Since it
    is harmless to do so, we can remove the WARN in badrange_forget.
    
    Cc: Dave Jiang <dave.jiang@intel.com>
    Cc: Dan Williams <dan.j.williams@intel.com>
    Signed-off-by: Vishal Verma <vishal.l.verma@intel.com>
    Signed-off-by: Dan Williams <dan.j.williams@intel.com>

diff --git a/drivers/nvdimm/badrange.c b/drivers/nvdimm/badrange.c
index 0b67dcf43234..e068d72b4357 100644
--- a/drivers/nvdimm/badrange.c
+++ b/drivers/nvdimm/badrange.c
@@ -114,7 +114,6 @@ void badrange_forget(struct badrange *badrange, phys_addr_t start,
 	struct badrange_entry *bre, *next;
 
 	spin_lock(&badrange->lock);
-	WARN_ON_ONCE(list_empty(badrange_list));
 
 	/*
 	 * [start, clr_end] is the badrange interval being cleared.